lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManager.java

```diff
@@ -487,6 +487,9 @@ public void onResourceLaunch(final ResourceLaunchEvent resourceLaunchEvent) {
       if (this.stateManager.isAllocated()) {
         this.stateManager.setSubmitted();
         this.resourceLaunchHandler.onNext(resourceLaunchEvent);
+      } else if (this.stateManager.isFailedOrKilled()) {
+        LOG.log(Level.WARNING, "Evaluator manager expected" + EvaluatorState.ALLOCATED +
```

Comment:
Ah, yes, that makes sense. The only call to `EvaluatorStatusManager.setFailed` is in `org.apache.reef.runtime.common.driver.evaluator.EvaluatorManager.onEvaluatorException`, which itself is called when we either receive an exception from the Evaluator itself or a shutdown notice from the RM. The latter is the case here.